web/assets/auto-apply-filter.js

```diff
@@ -0,0 +1,11 @@
+export default function autoApplyFilter() {
+    const filters = document.querySelectorAll("input.govuk-checkboxes__input");
+
+    filters.forEach((filter) => {
+        filter.addEventListener('click', () => {
+            setTimeout(() => {
+                document.forms["search-filters"].submit();
+            }, 500)
+        })
+    })
```

Comment:
I think we should add a buffer to the timeout, so that it will submit the form after they've not clicked anything for 500ms, rather than giving them exactly 500ms from the first click.
```suggestion
    const filters = document.querySelectorAll("input.govuk-checkboxes__input");
    let timeout = null;

    filters.forEach((filter) => {
        filter.addEventListener('click', () => {
            if (timeout !== null) clearTimeout(timeout);

            timeout = setTimeout(() => {
                document.forms["search-filters"].submit();
            }, 500)
        })
    })
```